app/src/shared/components/ActivityList/ActivityListItem/index.jsx

```diff
@@ -0,0 +1,236 @@
+    && a {
+        display: inline-block;
+        padding: 0;
+        background-color: transparent;
+        font-weight: 500;
+        line-height: 18px;
+    }
+
+    ${Container}:hover && a {
+        color: #0324FF;
+    }
+`
+
+const Details = styled.div`
+    color: #a3a3a3;
+    font-size: 12px;
+`
+
+const Type = styled.span`
+    color: #323232;
+`
+
+type Props = {
+    activity: Activity,
+    user?: any,
+    stream?: any,
+    product?: any,
+    canvas?: any,
+}
+
+const renderItem = (action: string, linkTitle: ?string, linkHref: ?string, id: string) => (
+    <React.Fragment>
+        {linkHref != null ? (
+            <a
+                href={linkHref}
+            >
+                {linkTitle}
+            </a>
+        ) : (
+            id
+        )}
+        &nbsp;
+        {I18n.t(`shared.action.${action.toLowerCase()}`)}
+    </React.Fragment>
+)
+
+const renderStreamItem = (id, stream, action) => (
+    renderItem(
+        action,
+        stream && stream.name,
+        stream && routes.userPageStreamShow({
+            streamId: stream.id,
+        }),
+        id,
+    )
+)
+
+const renderProductItem = (id, product, action) => (
+    renderItem(
+        action,
+        product && product.name,
+        product && routes.editProduct({
+            id: product.id,
+        }),
+        id,
+    )
+)
+
+const renderCanvasItem = (id, canvas, action) => (
+    renderItem(
+        action,
+        canvas && canvas.name,
+        canvas && routes.canvasEdit({
+            id: canvas.id,
+        }),
+        id,
+    )
+)
+
+const renderContent = (activity, stream, product, canvas) => {
+    if (activity.streamId != null) {
+        return renderStreamItem(activity.streamId, stream, activity.action)
+    }
+    if (activity.productId != null) {
+        return renderProductItem(activity.productId, product, activity.action)
+    }
+    if (activity.canvasId != null) {
+        return renderCanvasItem(activity.canvasId, canvas, activity.action)
+    }
+    return activity.action
+}
+
+const renderImage = (activity, user, stream, product, canvas, isLoading) => {
+    if (product) {
+        return <StyledAvatar alt={product.name} src={product.imageUrl} isLoading={isLoading} />
+    }
+    if (canvas) {
+        return <SvgIcon name="canvas" />
+    }
+    if (stream) {
+        return <SvgIcon name="stream" />
+    }
+    if (user) {
+        return <StyledAvatar alt={user.name} isLoading={isLoading} circle />
+    }
+    return null
+}
+
+const renderType = (stream, product, canvas) => {
+    if (product && product.type === productTypes.DATAUNION) {
+        return I18n.t('general.dataUnion')
+    }
+    if (product && product.type === productTypes.NORMAL) {
+        return I18n.t('general.dataProduct')
+    }
+    if (canvas) {
+        return I18n.t('general.canvas')
+    }
+    if (stream) {
+        return I18n.t('general.stream')
+    }
+    return null
+}
+
+const ActivityListItem = ({
```

Comment:
I wonder if it's really required to pass all them props to this component (user, stream, product, canvas). I know they're optional but wouldn't it be more convenient to have, let's say, `resource` and `resourceType`. Or just `resource` which type can be determined from within the `activity`. This could potentially simplify this component. Example:

```jsx
<ActivityListItem activity={…} resource={user /* or product/canvas/whatever */} />
```